build.sbt

```diff
@@ -1,96 +1,36 @@
-      "scm:git@github.com/raquo/scala-dom-types.git"
+ThisBuild / scalaVersion := ScalaVersions.v213
+ThisBuild / crossScalaVersions := Seq(ScalaVersions.v3RC1, ScalaVersions.v213, ScalaVersions.v212, ScalaVersions.v211)
+
+lazy val domtypes =
+  crossProject(JSPlatform, JVMPlatform).in(file("."))
+    .settings(
+      scalacOptions ~= (_.filterNot(Set(
+        "-Wunused:params",
+        "-Ywarn-unused:params",
+        "-Wunused:explicits"
+      ))),
+      scalacOptions in (Compile, doc) ~= (_.filter(_.startsWith("-Xplugin"))),
+      scalacOptions in (Compile, doc) ++= Seq(
+        "-no-link-warnings" // Suppress scaladoc "Could not find any member to link for" warnings
+      )
     )
-  ),
-  developers := List(
-    Developer(
-      id = "raquo",
-      name = "Nikita Gazarov",
-      email = "nikita@raquo.com",
-      url = url("http://raquo.com")
+    .jsSettings(
+      scalaJSLinkerConfig ~= { _.withSourceMap(false) },
+      libraryDependencies ++= Seq(
+        ("org.scala-js" %%% "scalajs-dom" % "1.1.0").withDottyCompat(scalaVersion.value)
+      )
     )
-  ),
-  sonatypeProfileName := "com.raquo",
-  publishMavenStyle := true,
-  publishArtifact in Test := false,
-  publishTo := sonatypePublishTo.value,
-  releaseCrossBuild := true,
-  pomIncludeRepository := { _ => false },
-  useGpg := false,
-  releasePublishArtifactsAction := PgpKeys.publishSigned.value
-)
-
 
-val baseScalacSettings =
-  "-encoding" :: "UTF-8" ::
-  "-unchecked" ::
-  "-deprecation" ::
-  "-explaintypes" ::
-  "-feature" ::
-  "-language:_" ::
-  "-Xfuture" ::
-  "-Xlint" ::
-  "-Yno-adapted-args" ::
-  "-Ywarn-value-discard" ::
-  "-Ywarn-unused" ::
-  Nil
-
-lazy val scalacSettings = Seq(
-  scalacOptions ++= {
-    CrossVersion.partialVersion(scalaVersion.value) match {
-      case Some((2, 13)) =>
-        baseScalacSettings.diff(
-          "-Xfuture" ::
-          "-Yno-adapted-args" ::
-          "-Ywarn-infer-any" ::
-          "-Ywarn-nullary-override" ::
-          "-Ywarn-nullary-unit" ::
-          Nil
-        )
-      case _ => baseScalacSettings
-    }
-  }
+lazy val noPublish = Seq(
+  publishLocal / skip := true,
+  publish / skip := true,
+  publishTo := Some(Resolver.file("Unused transient repository", file("target/unusedrepo")))
 )
 
-lazy val commonSettings = releaseSettings ++ scalacSettings
-
-lazy val root = project.in(file("."))
-  .aggregate(domtypesJS, domtypesJVM)
-  .settings(commonSettings)
-  .settings(
-    skip in publish := true
-  )
-
-lazy val domtypes = crossProject(JSPlatform, JVMPlatform).in(file("."))
-  .settings(commonSettings)
-  .jsConfigure(_.enablePlugins(ScalaJSBundlerPlugin))
-  .jsSettings(
-    scalaJSLinkerConfig ~= { _.withSourceMap(false) },
-    requireJsDomEnv in Test := true,
-    useYarn := true,
-    libraryDependencies ++= Seq(
-      "org.scala-js" %%% "scalajs-dom" % "1.0.0"
+lazy val root =
+  project.in(file("."))
+    .aggregate(domtypes.js, domtypes.jvm)
+    .settings(noPublish)
```

Comment:
I guess it's not possible to configure noPublish from release.sbt? Not sure which way is better, it just seems everything else release-related is there now.

Reply:
yeah, can't refer a `lazy val` defined in a different sbt file